tests/pw_actions_assertions/registrationForm_WithPOM.spec.js

```diff
@@ -0,0 +1,215 @@
+          await expect(signUpPopup.container).toHaveScreenshot('Name required.png');
+          await expect(signUpPopup.registerBtn).toBeDisabled();
+        });
+
+        await test.step('Invalid "Name" values should give an error', async () => {
+          for (const [fieldName, value] of Object.entries(testNameData)) {
+            await signUpPopup.nameField.fill(value);
+            await expect(signUpPopup.nameField, `Filled in value should be '${value}'`).toHaveValue(value);
+
+            if (fieldName === 'shortValue' || fieldName === 'longValue') {
+              await expect(signUpPopup.invalidField).toHaveText('Name has to be from 2 to 20 characters long');
+            } else if (fieldName === 'LongInvalidValue') {
+              await expect(signUpPopup.invalidField).toHaveText(/Name is invalid.*Name has to be from 2 to 20 characters long/);
+            } else {
+              await expect(signUpPopup.invalidField).toHaveText('Name is invalid');
+            }
+            await expect(signUpPopup.registerBtn).toBeDisabled();
+            await signUpPopup.nameField.clear();
+          }
+        });
+
+        await test.step('"Name" input field should have red border in case of error', async () => {
+          await expect(signUpPopup.invalidField).toHaveCSS('border-color', 'rgb(220, 53, 69)');
+        });
+      });
+    });
+
+    test.describe('"Last name" field', () => {
+      test('"Last name" field cases', async () => {
+        await test.step('Empty "Last name" value should give an error', async () => {
+          await signUpPopup.lastNameField.focus();
+          await signUpPopup.lastNameField.blur();
+          await expect(signUpPopup.invalidField).toHaveText('Last name required');
+          await expect(signUpPopup.registerBtn).toBeDisabled();
+        });
+
+        await test.step('Invalid "Last name" values should give an error', async () => {
+          for (const [fieldName, value] of Object.entries(testLastNameData)) {
+            await signUpPopup.lastNameField.fill(value);
+            await expect(signUpPopup.lastNameField, `Filled in value should be '${value}'`).toHaveValue(value);
+
+            if (fieldName === 'shortValue' || fieldName === 'longValue') {
+              await expect(signUpPopup.invalidField).toHaveText('Last name has to be from 2 to 20 characters long');
+            } else if (fieldName === 'LongInvalidValue') {
+              await expect(signUpPopup.invalidField).toHaveText(/Last name is invalid.*Last name has to be from 2 to 20 characters long/);
+              await expect(signUpPopup.container).toHaveScreenshot('Invalid and long Last name.png');
+            } else {
+              await expect(signUpPopup.invalidField).toHaveText('Last name is invalid');
+            }
+            await expect(signUpPopup.registerBtn).toBeDisabled();
+            await signUpPopup.lastNameField.clear();
+          }
+        });
```

Comment:
it loos a bit overcomplicated, I would recommend to make it simpler

Reply:
I'll update it after the lesson about fixtures and move expect result there